FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -125,18 +125,20 @@ - (void)installationIDWithCompletion:(FIRInstallationsIDHandler)completion {
 }
 
 - (void)authTokenWithCompletion:(FIRInstallationsTokenHandler)completion {
-  // TODO: Implement
-  FIRInstallationsAuthTokenResult *result =
-      [[FIRInstallationsAuthTokenResult alloc] initWithToken:@"token" expirationDate:[NSDate date]];
-  completion(result, nil);
+  [self authTokenForcingRefresh:NO completion:completion];
 }
 
 - (void)authTokenForcingRefresh:(BOOL)forceRefresh
                      completion:(FIRInstallationsTokenHandler)completion {
-  // TODO: Implement
-  FIRInstallationsAuthTokenResult *result =
-      [[FIRInstallationsAuthTokenResult alloc] initWithToken:@"token" expirationDate:[NSDate date]];
-  completion(result, nil);
+  [self.installationsIDController getAuthTokenForcingRefresh:forceRefresh]
```

Comment:
FYI, might also just cache it in the central place, but we will come back in later discussions.

Reply:
Yeah, I think in-memory cache on `FIRSecureStorage` level should solve the slow Keychain access question universally for any part that depends on it.